src/libstore/unix/build/local-derivation-goal.cc

```diff
@@ -2161,6 +2162,10 @@ void LocalDerivationGoal::runChild()
                     sandboxProfile += fmt("\t(subpath \"%s\")\n", path);
                 else
                     sandboxProfile += fmt("\t(literal \"%s\")\n", path);
+
+                if(sandboxProfile.length() - initialLen > (1 << 15)) {
+                    sandboxProfile += ")\n(allow file-read* file-write* process-exec\n";
+                }
```

Comment:
```suggestion
                if(sandboxProfile.length() > nextCutoff) {
                    sandboxProfile += ")\n(allow file-read* file-write* process-exec\n";
                    nextCutoff += cutoffInterval;
                }
```

Are empty one permissible?
I suppose you could add a hardcoded empty one to find out.